logistics/scripts/run-smoke-tests.ps1

```diff
@@ -114,42 +111,45 @@ function Get-SmokeTestConfiguration {
 
     $config.bulkLoadTempJsonConfig = Join-Path $env:temp "smokeTestconfig.json"
 
+    $config.buildConfiguration = "Debug"
+    if (-not [string]::IsNullOrWhiteSpace($env:msbuild_buildConfiguration)) { $config.buildConfiguration = $env:msbuild_buildConfiguration }
+    if (-not [string]::IsNullOrWhiteSpace($env:msbuild_exe)) { $config.msbuild_exe = $env:msbuild_exe }
+
     return $config
 }
 
-$script:tasks = @()
-$script:result = @()
 $errorCode = 0;
 
 try {
     Clear-Error
 
-    Set-TemplateConfigurationScript { Get-SmokeTestConfiguration }
-    (Get-TemplateConfiguration).GetEnumerator() | Sort-Object -Property Name | Format-Table -HideTableHeaders -AutoSize -Wrap
+    $config = Get-SmokeTestConfiguration
+    $config.GetEnumerator() | Sort-Object -Property Name | Format-Table -HideTableHeaders -AutoSize -Wrap
+
 
     if (-not $noRebuild) {
-        $tasks += @(
-            'Invoke-RestoreLoadToolsPackages'
-            'Invoke-BuildLoadTools'
-            'Invoke-SetTestHarnessConfig')
+        $script:result += Invoke-Task "Invoke-RestoreLoadToolsPackages" { Invoke-RestoreLoadToolsPackages $config }
+        $script:result += Invoke-Task "Invoke-BuildLoadTools" { Invoke-BuildLoadTools $config }
     }
 
-    $tasks += @(
-        'Add-RandomKeySecret'
-        'Invoke-StartTestHarness'
-        'Invoke-SmokeTestClient')
+    $script:result += Invoke-Task "Add-RandomKeySecret" { Add-RandomKeySecret $config }
 
-    $elapsed = Use-StopWatch {
-        foreach ($task in $tasks) {
-            $script:result += Invoke-Task -name $task -task { & $task }
+    $script:result += Invoke-Task "Start-TestHarness" {
```

Comment:
Now that we've broken the tasks up a bit, should the try move down to here?  No need for the finally to end the test harness unless we make it to the start test harness task.